src/tic_tac_toe/unbeatable_comp.clj

```diff
@@ -22,18 +22,22 @@
       (nil? winner) tie
       :else (* (- depth maximum-depth) perspective))))
 
-(declare choose-best-space)
+(defn- at-max-depth?
+  [depth]
+  (> -1 (- maximum-depth depth)))
+
+(declare get-tile-from-comp)
```

Comment:
Is there a reason you chose to use a `declare` statement over moving `simulate-next-move` further down your file to avoid the declaration?

Reply:
hi! yes in fact 'get-tile-from-comp' is the entry-point for the minimax (it's my 'choose-best-space') - and this cross-refers to simulate-next-move (because at some point the 'choose-best-move' gets to 'score-move' which itself refers to 'simulate-next-move') - so one of them needs to be declared